aiplatform/src/main/java/aiplatform/EmbeddingModelTuningSample.java

```diff
@@ -0,0 +1,127 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package aiplatform;
+
+// [START aiplatform_sdk_embedding_model_tuning]
+import com.google.cloud.aiplatform.v1.CreatePipelineJobRequest;
+import com.google.cloud.aiplatform.v1.LocationName;
+import com.google.cloud.aiplatform.v1.PipelineJob;
+import com.google.cloud.aiplatform.v1.PipelineJob.RuntimeConfig;
+import com.google.cloud.aiplatform.v1.PipelineServiceClient;
+import com.google.cloud.aiplatform.v1.PipelineServiceSettings;
+import com.google.protobuf.Value;
+import java.io.IOException;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+// [END aiplatform_sdk_embedding_model_tuning]
+
+public class EmbeddingModelTuningSample {
+  public static void main(String[] args) throws IOException {
+    // [START aiplatform_sdk_embedding_model_tuning]
+    // TODO(developer): Replace these variables before running this sample.
+    String apiEndpoint = "us-central1-aiplatform.googleapis.com:443";
+    String project = "PROJECT";
+    String baseModelVersionId = "BASE_MODEL_VERSION_ID";
+    String taskType = "TASK_TYPE";
+    String pipelineJobDisplayName = "PIPELINE_JOB_DISPLAY_NAME";
+    String outputDir = "OUTPUT_DIR";
+    String queriesPath = "QUERIES";
+    String corpusPath = "CORPUS";
+    String trainLabelPath = "TRAIN_LABEL";
+    String testLabelPath = "TEST_LABEL";
+    int batchSize = 128;
+    int iterations = 1000;
+
+    createEmbeddingModelTuningPipelineJob(
+        apiEndpoint,
+        project,
+        baseModelVersionId,
+        taskType,
+        pipelineJobDisplayName,
+        outputDir,
+        queriesPath,
+        corpusPath,
+        trainLabelPath,
+        testLabelPath,
+        batchSize,
+        iterations);
+    // [END aiplatform_sdk_embedding_model_tuning]
+  }
+
+  // [START aiplatform_sdk_embedding_model_tuning]
+  public static PipelineJob createEmbeddingModelTuningPipelineJob(
+      String apiEndpoint,
+      String project,
+      String baseModelVersionId,
+      String taskType,
+      String pipelineJobDisplayName,
+      String outputDir,
+      String queriesPath,
+      String corpusPath,
+      String trainLabelPath,
+      String testLabelPath,
+      int batchSize,
+      int iterations)
+      throws IOException {
+    Matcher matcher = Pattern.compile("^(?<Location>\\w+-\\w+)").matcher(apiEndpoint);
+    String location = matcher.matches() ? matcher.group("Location") : "us-central1";
```

Comment:
This is hard to understand what's going on. Instead of using a regular expression to extract the location, how about we simply pass the location as an extra parameter to the function? We can have an explicit comment that the location must match the endpoint used.

Otherwise, a simple split by `-` and getting the first item seems like a simpler solution than a regular expression.

Reply:
I have already implemented the same logic in python, go, and nodejs as well.
* Found this simple regex "^(\w+-\w+)" works for almost all the cases.

In Python, given a funny api_endpoint = "us-central1-funny-staging-strage-prediction-aiplatform.sandy.boxer.googleapis.com"
* location = "-".join(api_endpoint.split(".")[0].split("-")[:2])  # I found it is hairy to do two splits and a join.

Reply:
For consistency reasons, you would want this to be the same as other language samples.